src/aktualizr_secondary/secondary_tcp_server.cc

```diff
@@ -77,8 +77,8 @@ bool SecondaryTcpServer::HandleOneConnection(int socket) {
 
     do {
       received = recv(socket, buffer.Tail(), buffer.TailSpace(), 0);
-      LOG_TRACE << "Got " << received << " bytes "
-                << Utils::toBase64(std::string(buffer.Tail(), static_cast<size_t>(received)));
+      // LOG_TRACE << "Got " << received << " bytes "
+      //           << Utils::toBase64(std::string(buffer.Tail(), static_cast<size_t>(received)));
```

Comment:
I think the most apparent difference in database is the "device_info" table for secondary is empty. So maybe I can check it to know whether this is primary or secondary.

Reply:
> I think the most apparent difference in database is the "device_info" table for secondary is empty. So maybe I can check it to know whether this is primary or secondary.

I think if that's missing, it could also mean that provisioning has failed. Maybe if you check if that is absent but something else is present that would suffice?

Reply:
Ah, yes. I think for now, we can check "tls_creds", it's only used in primary application. Do we have any plan to change this behaviour ?

Reply:
`tls_creds` might be even better than `device_info`, true. I don't expect that behavior will change. But I think we still might need to check if one or the other is empty *and* something else is non-empty to determine if it's a Secondary.